### PR TITLE
Use @react.component for ErrorBoundary

### PR DESCRIPTION
**Pull request by mattdamon108**
I've rewrite `RescriptReactErrorBoundary` component implementation with `@react.component`. It is written without it, then it is vulnerable to being broken in case JSX PPX output is changed. Related [discussion](https://github.com/rescript-lang/syntax/pull/547#issuecomment-1159857861) in the PR of JSX V4 of syntax repo.

### CODE REVIEW

**Review by cristianoc — APPROVED** [review]
Thanks!
Would you update the changelog?

File: Changes.md [hunk 1]
```diff
@@ -26,12 +26,14 @@ For history on previous evolution of the code, check out the original [reason-re
 - **IMPORTANT:** Currently, old third-party packages that are still dependent on `reason-react` will not mix with other `@rescript/react` based code due to a build system problem. Which means that every third-party dependency needs to be upgraded to `@rescript/react` first to make it compatible. See [this forum discussion](https://forum.rescript-lang.org/t/discussion-reason-react-rescript-react-migration-path/1086) for more details.
 
 - Removed legacy modules ("record api"):
+
```
Comment (mattdamon108) [inline Changes.md:29]: Oops, unnecessary formatting. Let me know if you want me to remove this.

Comment (cristianoc) [reply Changes.md:29]: Looks like it's formatted correctly now. And it was not before.
